Assignment 5 qc.Rmd

```diff
@@ -0,0 +1,86 @@
+First we will build a classification tree to predict which students ask a teacher for help, which start a new session, or which give up, based on whether or not the student completed a session (D1$complete) and whether or not they asked for hints (D1$hint.y). 
+```{r}
+
+c.tree <- rpart(action ~ hint.y + complete, method="class", data=D1) #Notice the standard R notion for a formula X ~ Y
+
+#Look at the error of this tree
+printcp(c.tree)
+
+#Plot the tree
+post(c.tree, file = "tree.ps", title = "Session Completion Action: 1 - Ask teacher, 2 - Start new session, 3 - Give up")
+
+```
+## Part II
+
+#Regression Tree
+
+We want to see if we can build a decision tree to help teachers decide which students to follow up with, based on students' performance in Assistments. We will create three groups ("teacher should intervene", "teacher should monitor student progress" and "no action") based on students' previous use of the system and how many hints they use. To do this we will be building a decision tree using the "party" package. The party package builds decision trees based on a set of statistical stopping rules.
+
+#Take a look at our outcome variable "score"
+```{r}
+hist(D1$score)
+```
+
+#Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
+```{r}
+D1$advice <- ifelse(D1$score <=0.4, "intervene", ifelse(D1$score > 0.4 & D1$score <=0.8, "monitor", "no action"))
+```
+
+#Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
+```{r}
+score_ctree <- ctree(factor(advice) ~ prior_prob_count + prior_percent_correct + hints, D1)
+```
+
+#Plot tree
+```{r}
+plot(score_ctree)
+```
+
+Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
+```{r}
+score_ctree
+```
+A:Hints and prior_prob_count.From above we know that as the root node hints has the largest information benefit so it's the most important variable.When it comes to prior_prob_count and prior_percent_correct,the formal has larger weights and if we count the information benefit we will find that the formal is more important than the latter.
+
+#Test Tree
+Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
+
+```{r}
+#Upload new data
+
+D2 <- read.csv('intelligent_tutor_new.csv')
+
+#Generate predicted advice for new students based on tree generated from old students
+
+D2$prediction <- predict(score_ctree, D2)
+
+``` 
+## Part III
+Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+```{r}
+table(D2$prediction,D2$score)
```

Comment:
You could also calculate this from the dataframe: `mean(ifelse(D2$prediction == "no action", 1, 0))`